### PR TITLE
Preserve test names in test-quarantine data

### PR DESCRIPTION
**Pull request by wtgodbe**
The Part 1 safety scrub was applied to the complete serialized JSON payload. Its broad token fallback also matched long test method names, replacing multiple identifiers with `[REDACTED]` and creating duplicate JSON keys. This caused unrelated failures to be combined and led to the incorrect quarantine in #68727.

This change:
- keeps secret scrubbing scoped to captured error messages, stack traces, and Helix failure blocks
- preserves structured test-name keys during serialization
- validates the emitted JSON and fails closed if duplicate keys are ever introduced

The focused regression check uses the two test names that collided in workflow run 32632851798 and confirms the old whole-payload scrub creates a duplicate while the new path preserves both identifiers.

The workflow source and generated lockfile were compiled successfully with `gh aw compile test-quarantine --strict`.

### CODE REVIEW

File: .github/workflows/test-quarantine.md [hunk 1]
```diff
@@ -1106,12 +1108,30 @@ on:
...
+                for key, value in pairs:
+                    if key in result:
+                        duplicates.add(key)
+                    result[key] = value
+                return result
+
+            json.loads(js, object_pairs_hook=reject_duplicate_keys)
+            if duplicates:
+                examples = ", ".join(sorted(duplicates)[:3])
+                sys.exit(f"FATAL: part1_data contains duplicate JSON keys: {examples}")
```
[the remark this answers is not in the document]
Comment (wtgodbe) [reply .github/workflows/test-quarantine.md]: Added an explicit `json.JSONDecodeError` handler that exits with `FATAL: part1_data is invalid JSON: ...`. The regenerated lockfile includes the same handling.

File: .github/workflows/test-quarantine.lock.yml [hunk 2]
```diff
@@ -2515,7 +2535,7 @@ jobs:
           GH_AW_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
           GH_AW_AGENT_CONCLUSION: ${{ needs.agent.result }}
           GH_AW_WORKFLOW_ID: "test-quarantine"
-          GH_AW_ACTION_FAILURE_ISSUE_EXPIRES_HOURS: "0"
+          GH_AW_ACTION_FAILURE_ISSUE_EXPIRES_HOURS: "168"
```
[the remark this answers is not in the document]
Comment (wtgodbe) [reply .github/workflows/test-quarantine.lock.yml]: Restored the existing `0` value after recompiling so the PR no longer includes the unrelated action-failure expiry drift. The source already has `safe-outputs.report-failure-as-issue: false`; this was not an intended behavior change.

[the remark this answers is not in the document]
Comment (wtgodbe) [reply .github/workflows/test-quarantine.lock.yml]: Correction: I should not have manually restored a generated lockfile value. I reran `gh aw compile test-quarantine --strict` and pushed the lockfile exactly as emitted in fd04daba75. The 168-hour value is compiler output; failure issue reporting remains disabled by the source configuration.
